pkg/sdkserver/sdkserver_test.go

```diff
@@ -678,176 +685,185 @@ func TestSDKServerUpdateEventHandler(t *testing.T) {
-			return true, gs, nil
-		})
+	close(stop)
+	wg.Wait()
+}
 
-		_, err := sc.Allocate(context.Background(), &sdk.Empty{})
-		assert.NoError(t, err)
+func TestSDKServerReserveTimeout(t *testing.T) {
+	t.Parallel()
+	m := agtesting.NewMocks()
 
-		select {
-		case <-done:
-		case <-time.After(5 * time.Second):
-			assert.Fail(t, "have contention")
-		}
+	state := make(chan v1alpha1.GameServerStatus, 100)
+	defer close(state)
 
-		select {
-		case <-done:
-		case <-time.After(5 * time.Second):
-			assert.Fail(t, "should be updated")
+	m.AgonesClient.AddReactor("list", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		gs := v1alpha1.GameServer{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test", Namespace: "default",
+			},
+			Spec: v1alpha1.GameServerSpec{Health: v1alpha1.Health{Disabled: true}},
 		}
+		gs.ApplyDefaults()
+		return true, &v1alpha1.GameServerList{Items: []v1alpha1.GameServer{gs}}, nil
 	})
 
-	t.Run("timeout", func(t *testing.T) {
-		m, sc, stop := setup(t, defaultGs.DeepCopy())
-		defer close(stop)
+	m.AgonesClient.AddReactor("update", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		ua := action.(k8stesting.UpdateAction)
+		gs := ua.GetObject().(*v1alpha1.GameServer)
 
-		now := time.Now()
-		fc := clock.NewFakeClock(now)
-		sc.clock = fc
+		state <- gs.Status
 
-		done := make(chan struct{}, 10)
+		return true, gs, nil
+	})
 
-		m.AgonesClient.AddReactor("update", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
-			defer func() {
-				done <- struct{}{}
-			}()
+	sc, err := defaultSidecar(m)
 
-			// move past timeout
-			fc.Step(40 * time.Second)
-			return true, nil, k8serrors.NewConflict(schema.ParseGroupResource(""), "gameservers", errors.New("contention"))
-		})
+	assert.NoError(t, err)
+	stop := make(chan struct{})
+	sc.informerFactory.Start(stop)
+	assert.True(t, cache.WaitForCacheSync(stop, sc.gameServerSynced))
 
-		_, err := sc.Allocate(context.Background(), &sdk.Empty{})
-		assert.EqualError(t, err, "Allocation request timed out")
+	wg := sync.WaitGroup{}
+	wg.Add(1)
 
+	go func() {
+		err = sc.Run(stop)
+		assert.Nil(t, err)
+		wg.Done()
+	}()
+
+	assertStateChange := func(expected v1alpha1.GameServerState, timeout time.Duration, additional func(status v1alpha1.GameServerStatus)) {
 		select {
-		case <-done:
-		case <-time.After(5 * time.Second):
-			assert.Fail(t, "have contention")
+		case current := <-state:
+			assert.Equal(t, expected, current.State)
+			additional(current)
+		case <-time.After(timeout):
+			assert.Fail(t, "should have gone to Reserved by now")
 		}
-	})
+	}
+	noAdditional := func(status v1alpha1.GameServerStatus) {}
 
-	t.Run("unhealthy gameserver", func(t *testing.T) {
-		gs := defaultGs.DeepCopy()
-		gs.Status.State = v1alpha1.GameServerStateUnhealthy
-		_, sc, stop := setup(t, gs.DeepCopy())
-		defer close(stop)
+	// 3 seconds
```

Comment:
This comment doesn't add much value, since the 3 seconds is obvious from the line below. Is there something worth describing here? If not, please remove.